crates/theme/src/settings.rs

```diff
@@ -52,13 +109,67 @@ pub struct ThemeSettingsContent {
     #[serde(default)]
     pub theme: Option<String>,
 
+    /// Set the themes to use for light and dark mode. Note that this
+    /// Supersedes the `theme` setting.
+    #[serde(default)]
+    pub themes: Option<ThemeFeatures>,
+
     /// EXPERIMENTAL: Overrides for the current theme.
     ///
     /// These values will override the ones on the current theme specified in `theme`.
     #[serde(rename = "experimental.theme_overrides", default)]
     pub theme_overrides: Option<ThemeStyleContent>,
 }
 
+impl ThemeSettingsContent {
+    /// Returns whether the current theme setting requires to use the dark theme.
+    pub(crate) fn is_dark_mode(&self, system_is_dark: bool) -> bool {
+        if let Some(themes) = &self.themes {
+            return themes.is_dark_mode(system_is_dark);
+        }
+
+        false
+    }
+
+    pub(crate) fn theme_to_use(&self, system_is_dark: bool) -> Option<&str> {
+        if let Some(themes) = &self.themes {
+            if self.is_dark_mode(system_is_dark) {
+                if let Some(theme) = &themes.dark {
+                    return Some(theme.as_str());
+                }
+            } else {
+                if let Some(theme) = &themes.light {
+                    return Some(theme.as_str());
+                }
+            }
+        }
+
+        self.theme.as_deref()
+    }
+}
+
+impl ThemeFeatures {
+    pub fn is_dark_mode(&self, system_is_dark: bool) -> bool {
+        match self.mode {
+            Some(ThemeMode::Auto) => system_is_dark,
+            Some(ThemeMode::Dark) => true,
+            Some(ThemeMode::Light) | _ => false,
```

Comment:
Any reason to not just use `None` here instead of a wildcard? It looks like that should be the only missing case:

```suggestion
            Some(ThemeMode::Light) | None => false,
```